demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+                                                                           strlen( sampleazureiotCOMMAND_START_TIME ),
+                                                                           ucCommandStartTimeValueBuffer, ulSinceTimeLength ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending start time: result 0x%08x", xResult ) );
+    }
+    /* Faking the end time to simplify dependencies on <time.h> */
+    else if( ( xResult = AzureIoTJSONWriter_AppendPropertyWithStringValue( pxWriter, ( const uint8_t * ) sampleazureiotCOMMAND_END_TIME,
+                                                                           strlen( sampleazureiotCOMMAND_END_TIME ),
+                                                                           ( const uint8_t * ) sampleazureiotCOMMAND_FAKE_END_TIME,
+                                                                           strlen( sampleazureiotCOMMAND_FAKE_END_TIME ) ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending end time: result 0x%08x", xResult ) );
+    }
+    else if( ( xResult = AzureIoTJSONWriter_AppendEndObject( pxWriter ) )
+             != eAzureIoTSuccess )
+    {
+        LogError( ( "Error appending end object: result 0x%08x", xResult ) );
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Command message callback handler
+ */
+static void prvHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                              void * pvContext )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lCommandNameLength;
+    int32_t ulCommandPayloadLength;
+    AzureIoTHubClient_t * pxHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    lCommandNameLength = strlen( sampleazureiotCOMMAND_MAX_MIN_REPORT );
+
+    if( ( lCommandNameLength == pxMessage->usCommandNameLength ) &&
+        ( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, ( const char * ) pxMessage->pucCommandName, lCommandNameLength ) == 0 ) )
+    {
+        /* Is for max min report */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        xResult = AzureIoTJSONWriter_Init( &xWriter, ucCommandPayloadBuffer, sizeof( ucCommandPayloadBuffer ) );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        xResult = prvInvokeMaxMinCommand( &xReader, &xWriter );
+
+        if( xResult == eAzureIoTSuccess )
+        {
+            ulCommandPayloadLength = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+            if( ulCommandPayloadLength > 0 )
+            {
+                if( AzureIoTHubClient_SendCommandResponse( pxHandle, pxMessage, 200,
```

Comment:
I recommend on `AzureIoTHubClient_SendCommandResponse` success - here and in the 501 and 404 responses below - that we have a LogInfo level that this has happened.  If a device is mysteriously not working and not able to be debugged live, logging "important" events like sample/SDK at least thinking it has sent a response can aid debug effort.